src/prefect/tasks/neo4j/neo4j_tasks.py

```diff
@@ -0,0 +1,243 @@
+            - `prefect.engine.signals.FAIL` if any error occurs while establishing
+                the connection with Neo4j.
+            - `prefect.engine.signals.FAIL` if any error occurs while running
+                the Cypher query.
+        """
+        if not server_uri and not server_uri_env_var:
+            msg = "Please provide either the `server_uri` or the `server_uri_env_var`."
+            raise ValueError(msg)
+
+        if not server_uri and server_uri_env_var not in os.environ.keys():
+            msg = f"`{server_uri_env_var}` not found in environment variables."
+            raise ValueError(msg)
+
+        neo4j_uri = server_uri or os.environ[server_uri_env_var]
+
+        if not user and not user_env_var:
+            msg = "Please provide either the `user` or the `user_env_var`."
+            raise ValueError(msg)
+
+        if not user and user_env_var not in os.environ.keys():
+            msg = f"`{user_env_var}` not found in environment variables."
+            raise ValueError(msg)
+
+        neo4j_user = user or os.environ[user_env_var]
+
+        if not password and not password_env_var:
+            msg = "Please provide either the `password` or the `password_env_var`."
+            raise ValueError(msg)
+
+        if not password and password_env_var not in os.environ.keys():
+            msg = f"`{password_env_var}` not found in environment variables."
+            raise ValueError(msg)
+
+        neo4j_password = password or os.environ[password]
+
+        neo4j_db_name = None
+        if db_name:
+            neo4j_db_name = db_name
+        elif db_name_env_var and db_name_env_var not in os.environ.keys():
+            msg = f"`{db_name_env_var}` not found in environment variables."
+            raise ValueError(msg)
+        elif db_name_env_var and db_name_env_var in os.environ.keys():
+            neo4j_db_name = os.environ[db_name_env_var]
+
+        if not cypher_query:
+            raise ValueError("Please provide a value for `cypher_query`.")
+
+        if return_result_as not in self.__ACCEPTED_RETURN_RESULT_TYPES:
+            msg = f"Illegal value for `return_result_as`. Illegal value is: {return_result_as}."
+            raise ValueError(msg)
+
+        try:
+            graph = Graph(
+                profile=neo4j_uri, name=neo4j_db_name, auth=(neo4j_user, neo4j_password)
+            )
+        except ConnectionUnavailable as e:
+            msg = f"Error while connecting to Neo4j. Exception: {str(e)}"
+            raise FAIL(message=msg)
+
+        try:
+            r = graph.run(cypher_query)
+        except ClientError as e:
+            msg = f"Error while running Cypher query. Exception: {str(e)}"
+            raise FAIL(message=msg)
+
+        result = r.data()
+
+        if not result:
+            return None
+        elif return_result_as == "json":
+            return dumps(result)
+        elif return_result_as == "dataframe":
+            return r.to_data_frame()
```

Comment:
Need to handle `return_result_as == "raw"`
```suggestion
        result = r.data()

        if not result:
            return None
        elif return_result_as == "json":
            return dumps(result)
        elif return_result_as == "dataframe":
            return r.to_data_frame()
        else:
            return result
```